src/lib/pickles_types/plonk_types.ml

```diff
@@ -1526,20 +1592,43 @@ module Messages = struct
     [@@deriving hlist, fields]
   end
 
-  let typ (type n f)
-      (module Impl : Snarky_backendless.Snark_intf.Run with type field = f) g
+  let typ (type n) g
       ({ runtime_tables; uses_lookups; lookups_per_row_4; _ } :
         Opt.Flag.t Features.Full.t ) ~dummy
-      ~(commitment_lengths : (((int, n) Vector.t as 'v), int, int) Poly.t) ~bool
-      =
-    let open Snarky_backendless.Typ in
+      ~(commitment_lengths : (((int, n) Vector.t as 'v), int, int) Poly.t) =
+    let module Impl = Kimchi_pasta_snarky_backend.Step_impl in
+    let open Impl.Typ in
+    let { Poly.w = w_lens; z; t } = commitment_lengths in
+    let array ~length elt = padded_array_typ ~dummy ~length elt in
+    let wo n = array ~length:(Vector.reduce_exn n ~f:Int.max) g in
+    let _w n =
```

Comment:
Do you need this? It seems to warn me that `With_degree_bound.typ` is unused otherwise. Let's see later. Maybe it is temporary?